packages/web-components/lib/components/selfie/src/SelfieCaptureScreens.js

```diff
@@ -233,40 +236,59 @@ class SelfieCaptureScreens extends HTMLElement {
   }
 
   // Force remount of selfie-capture-wrapper component for clean state
-  forceWrapperRemount() {
+  async forceWrapperRemount() {
+    SmartCamera.stopMedia();
+
     this._remountKey++;
+
     const container = this.querySelector('div');
-    const oldSelfieCapture = this.selfieCapture;
-
-    // Create new selfie-capture-wrapper element with same attributes
-    const newSelfieCapture = document.createElement('selfie-capture-wrapper');
-    newSelfieCapture.setAttribute('theme-color', this.themeColor);
-    newSelfieCapture.setAttribute('key', this._remountKey.toString());
-    newSelfieCapture.setAttribute('start-countdown', 'false');
-
-    // Copy all the correct attributes based on the getter logic
-    const showNav = this.hasAttribute('show-navigation');
-    const allowAgent = this.getAttribute('allow-agent-mode') === 'true';
-    const allowAgentTests = this.hasAttribute('show-agent-mode-for-tests');
-    const hideAttr = this.hasAttribute('hide-attribution');
-    const disableTests = this.hasAttribute('disable-image-tests');
-
-    if (showNav) newSelfieCapture.setAttribute('show-navigation', '');
-    if (allowAgent) newSelfieCapture.setAttribute('allow-agent-mode', 'true');
-    if (allowAgentTests)
-      newSelfieCapture.setAttribute('show-agent-mode-for-tests', '');
-    if (hideAttr) newSelfieCapture.setAttribute('hide-attribution', '');
-    if (disableTests) newSelfieCapture.setAttribute('disable-image-tests', '');
-    newSelfieCapture.setAttribute('hidden', '');
-
-    // Replace old element with new one
-    container.replaceChild(newSelfieCapture, oldSelfieCapture);
-
-    // Update reference
-    this.selfieCapture = newSelfieCapture;
-
-    // Re-add event listeners for the new element
-    this.setupSelfieWrapperEventListeners();
+    const oldWrapper = this.selfieCapture;
+
+    if (oldWrapper && container) {
+      // recreate wrapper element
+      oldWrapper.remove();
+      const newWrapper = document.createElement('selfie-capture-wrapper');
+
+      newWrapper.setAttribute('theme-color', this.themeColor);
+      newWrapper.setAttribute('key', this._remountKey.toString());
+      newWrapper.setAttribute('start-countdown', 'false');
+      newWrapper.setAttribute('hidden', '');
+
+      // copy all the attributes
+      if (this.hasAttribute('show-navigation')) {
+        newWrapper.setAttribute('show-navigation', '');
+      }
+      if (this.getAttribute('allow-agent-mode') === 'true') {
+        newWrapper.setAttribute('allow-agent-mode', 'true');
+      }
+      if (this.hasAttribute('show-agent-mode-for-tests')) {
+        newWrapper.setAttribute('show-agent-mode-for-tests', '');
+      }
+      if (this.hasAttribute('hide-attribution')) {
+        newWrapper.setAttribute('hide-attribution', '');
+      }
+      if (this.hasAttribute('disable-image-tests')) {
+        newWrapper.setAttribute('disable-image-tests', '');
+      }
```

Comment:
if we do need to transfer all the attributes, we can use `getAttributes` to retrieve the old wrapper attributes and map to the new wrapper